src/parser/mod.rs

```diff
@@ -1108,7 +1108,7 @@ impl<'a> Parser<'a> {
             if dialect_of!(self is PostgreSqlDialect | GenericDialect) =>
                 {
                     Ok(Some(Expr::Function(Function {
-                        name: ObjectName(vec![w.to_ident(w_span)]),
+                        name: ObjectName(vec![w.clone().into_ident(w_span)]),
```

Comment:
in this case I couldn't figure out (yet) how to avoid this clone given that `&w` is passed in

This PR doesn't increase the number of clones done, but it makes it more explicit when they are happening